src/Autosuggest.js

```diff
@@ -18,6 +20,12 @@ export default class Autosuggest extends Component { // eslint-disable-line no-s
     id: PropTypes.string                    // Used in aria-* attributes. If multiple Autosuggest's are rendered on a page, they must have unique ids.
   }
 
+  componentWillReceiveProps(nextProps) {
+    if (typeof nextProps.value === 'string') {
+      this.onUpdate(nextProps.value);
+    }
```

Comment:
@moroshko I added this check not to really ensure typing, but to make sure we don't trigger the `onUpdate` method when `props.value` is `null` or `undefined`—because that can cause broken behavior in the uncontrolled case. So, throwing an error here is not a good idea because it will likewise break people trying to use the uncontrolled behavior.

Perhaps this is less confusing?

```js
if (nextProps.value) {
  // If this component is controlled, then handle the value update
  this.onUpdate(nextProps.value);
}
```